application/app/views/widgets/_dataverse_dataset_versions.html.erb

```diff
@@ -0,0 +1,46 @@
+<%
+  dataverse_url = params[:dataverse_url]
+  persistent_id = params[:persistent_id]
+  repo_info = RepoRegistry.repo_db.get(dataverse_url)
+  api_key = repo_info&.metadata&.auth_key
+  service = Dataverse::DatasetService.new(dataverse_url, api_key: api_key)
+  begin
+    res = service.dataset_versions_by_persistent_id(persistent_id)
+    versions = res&.versions || []
+  rescue => e
+    versions = []
+  end
+  dv_url = Dataverse::DataverseUrl.parse(dataverse_url)
+%>
+<% if versions.any? %>
+  <ul class="list-group">
+    <% versions.each do |v| %>
+      <% version_path = view_dataverse_dataset_path(
+          dv_hostname: dv_url.domain,
+          persistent_id: persistent_id,
+          dv_scheme: dv_url.scheme_override,
+          dv_port: dv_url.port,
+          version: v.version
+        ) %>
+      <li class="list-group-item list-group-item-action p-0">
+        <%= link_to version_path, class: 'd-flex justify-content-between align-items-center gap-3 p-2 text-decoration-none' do %>
+          <div class="flex-grow-1 d-flex align-items-center gap-3">
+            <%= render partial: '/shared/file_row_date',
+                       locals: { date: v.create_time, title: t('dataverse.datasets.dataset_versions_widget.field_creation_date_title') } %>
+            <span class="badge text-bg-info"><%= v.version %></span>
+          </div>
+          <span class="btn btn-sm btn-outline-secondary">
+            <i class="bi bi-display" aria-hidden="true"></i>
+          </span>
+          <span class="visually-hidden"><%= t('dataverse.datasets.dataset_versions_widget.button_browse_version_label') %></span>
+        <% end %>
+      </li>
+    <% end %>
+  </ul>
+<% else %>
+  <div class="list-group list-group-flush">
+    <span class="list-group-item list-group-item-action" role="status">
+      <%= t('dataverse.datasets.dataset_files.msg_empty_dataset_text') %>
+    </span>
+  </div>
+<% end %>
```

Comment:
(1) I'm more in favor of calling the service classes from the controller and not directly from the view, but if this is for the sake of implementing lazy loading of the dataset versions it is OK.

(2) Nonetheless, since this view component is dataverse-specific, I would add it to the subfolder `widgets/dataverse` or inside `connectors/dataverse`

Reply:
Yes - I will move it to the DatasetController